baseplate/thrift/message_queue/message_queue.thrift

```diff
@@ -0,0 +1,29 @@
+namespace go reddit.message_queue
+namespace py message_queue.thrift
+namespace java com.reddit.baseplate.message_queue
+
+struct PutResponse {}
+
+struct GetResponse {
+    1: binary value;
+}
+
+exception TimedOutError {}
+
+service RemoteMessageQueueService {
+    PutResponse put(
+        1: string queue_name
+        2: i64 max_messages
```

Comment:
I know we need the `max_messages` value in the `RemoteMessageQueueHandler` `put` and `get` methods as they will create the queue if it doesn't exist. But that feels wrong to me. A queue put/get method call shouldn't need to specify the max size of the queue in every call. 

You seem to use the `create_queue` function everywhere anyway, so is it likely that the put/get methods will be called on a non-existant queue? Could you not specify a default queue size in the handler?

Reply:
Well, the client/baseplate user doesnt actually have to supply the queue size when calling get/put, that's done internally. When the `RemoteMessageQueue` queue is instantiated by the client, we don't connect to the server yet - we just store the parameters given. Then when get/put is called by the client, it connects and calls get/put on the server (these are the only two methods I expose from the server) and tacks on the `max_messages` parameter. Then the server handles creating the queue if necessary. Alternatively I can expose `create_server` and have the client do that, I dont really have a strong opinion.

I could specify a default, but the posix queue type doesn't, and I didnt want different behavior between the two.

Reply:
Ah, it is confusing now that I look again. `create_queue` does exist on the client side to instantiate the queue on the client side (it's a helper I created to more easily test that it was creating the correct queue type), but as mentioned it doesnt actually connect to the server yet